### DISCUSSION TITLE
Does useSWR always refetch (revalidate) the data? If so it cannot replace Redux?

### DISCUSSION THREAD

**Discussion by mimoid-prog** [vercel/swr#827] [Q&A] [answered]
I wanted to use SWR instead of Redux but I see one issue there.
As it is written in the docs "SWR is a strategy to first return the data from cache (stale), then send the fetch request (revalidate), and finally come with the up-to-date data.".
So SWR firstly takes cached data and even though it took cached data it gonna make request to the api to get the newest data, right? Then it makes no sense to use it instead of Redux because it always makes additional request?

**Comment by mimoid-prog** [comment]
Actually it can be closed. 
SWR is using cached data only for 2 seconds without revalidating. After that time revalidation happens and the fresh data is being fetched.
This value can be changed in config by `dedupingInterval`.
So I guess it's pretty good to use instead of Redux.

**Comment by shuding** [comment, marked as answer]
> Then it makes no sense to use it instead of Redux because it always makes additional request?

Why? SWR keeps the data refresh, and it will notify your UI. That's nothing to do with Redux (you can keep your data refresh with Redux too). If what you mean is local data mutation, check https://swr.vercel.app/docs/mutation.
